### PR TITLE
Add PyroFishingPro fishing and projectile launching task type

### PR DESCRIPTION
**Pull request by Krakenied**
Closes https://github.com/LMBishop/Quests/issues/571

### CODE REVIEW

**Comment by dinobossytnew** [comment]
@Krakenied hello bro dev pyro add coments: 
image
Captura

**Comment by Krakenied** [comment]
Yeah, however the shown event is still not a part of PyroFishingPro (Pyro hasn't released the shown version yet) so we are unable to make any use of it. And since we aren't, I've made just an alternative temporary solution for that (which surely works perfectly, it's even used a lot in tasks of quests on my server). Having considered all the possible solutions for that, it's the best one.

Obviously, I'm willing to improve the task type implementation in the future (as soon as Pyro releases the changes shown). Nevertheless the new version hasn't been released yet, so the only thing you can do for now is use the task type implementation available and hurry Pyro with release of the update.

**Comment by PyroTempus** [comment]
@Krakenied I have this update scheduled for Saturday (18th November), and this will be released as part of that update. 

As a note - from looking at your implementation of grabbing the ItemStack when it spawns on the player, that works in the majority of cases - however, quite a few servers use force-inventory which doesn't drop the item on the ground. (which from what I can see, would not be detected by this method). 

As a general implementation, this looks and should work fine for the majority use-case, but can be replaced by the new API method in the next patch. If there is anything else you need aside from the tier, and fishNumber, let me know and I'll see if I can attach it to this event until the full API is created.

**Comment by Krakenied** [comment]
@PyroTempus tier and fish number are enough

**Comment by Krakenied** [comment]
temporary changed to a draft

**Comment by Krakenied** [comment]
rebased

**Comment by Krakenied** [comment]
new task type example usage
```
tasks:
  pyroaaaaaa:
    type: "pyrofishingpro_fishing"
    fish-number: 1
    tier: Bronze
    amount: 69420

**Comment by Krakenied** [comment]
```
  aaaaaaaaaaaaaaaaaaa:
    type: "projectilelaunching"
    projectile: EGG
    amount: 500
  aaaaaaaaaaaaaaaaaaa2:
    type: "projectilelaunching"
    projectiles:
    - EGG
    - SNOWBALL
    amount: 500
```

**Comment by dinobossytnew** [comment]
ok =)

File: bukkit/libs/PyroFishingPro-4.8.28.jar
Comment (LMBishop) [inline bukkit/libs/PyroFishingPro-4.8.28.jar:1]: Will hosting the jar in this repository cause any issues with its author?

**Review by LMBishop — COMMENTED** [review]
Aside from the one concern below, looks good to me :)

File: bukkit/libs/PyroFishingPro-4.8.28.jar
Comment (Krakenied) [reply bukkit/libs/PyroFishingPro-4.8.28.jar:1]: Its only API

Comment (PyroTempus) [reply bukkit/libs/PyroFishingPro-4.8.28.jar:1]: > Will hosting the jar in this repository cause any issues with its author?

If it's just the API (which it appears to be) - that's completely fine with me.

Comment (LMBishop) [reply bukkit/libs/PyroFishingPro-4.8.28.jar:1]: Ah okay, awesome, thanks both

**Review by LMBishop — APPROVED** [review]

**Comment by SirCraftypants** [comment]
Screenshot 2024-01-16 102034
 Isn't working in the latest Quest version

**Comment by 00Hrtz** [comment]
> new task type example usage
> 
> ```
> tasks:
>   pyroaaaaaa:
>     type: "pyrofishingpro_fishing"
>     fish-number: 1
>     tier: Bronze
>     amount: 69420
> ```

I'm hoping you'll say yes when i ask if multiple fish numbers can be used i.e.

```
   pyroaaaaaa:
     type: "pyrofishingpro_fishing"
     fish-number: 1,2,3,4,5
     tier: Bronze
     amount: 69420

**Pull request merged**